test/unit/org/apache/cassandra/index/sai/analyzer/EqualsSupportTest.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.analyzer;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.apache.cassandra.index.sai.cql.VectorTester;
+
+/**
+ * Tests for {@link EqualsSupport}.
+ */
+public class EqualsSupportTest extends VectorTester
+{
+    @Before
+    public void populate()
+    {
+        createTable("CREATE TABLE %s (k int PRIMARY KEY, v text)");
+        execute("INSERT INTO %s (k, v) VALUES (1, 'Quick fox')");
+        execute("INSERT INTO %s (k, v) VALUES (2, 'Lazy fox')");
+    }
+
+    @Test
+    public void invalidOptions()
+    {
+        assertInvalidMessage(EqualsSupport.WRONG_OPTION_ERROR + "WRONG",
+                             "CREATE CUSTOM INDEX ON %s(v) USING 'StorageAttachedIndex' " +
+                             "WITH OPTIONS = {'case_sensitive': 'false', 'equals_behaviour_when_analyzed': 'WRONG'}");
+        assertInvalidMessage(EqualsSupport.WRONG_OPTION_ERROR + "WRONG",
+                             "CREATE CUSTOM INDEX ON %s(v) USING 'StorageAttachedIndex' " +
+                             "WITH OPTIONS = {'index_analyzer': 'standard', 'equals_behaviour_when_analyzed': 'WRONG'}");
+    }
+
+    @Test
+    public void nonAnalyzedDefault()
+    {
+        createIndex("CREATE CUSTOM INDEX ON %s(v) USING 'StorageAttachedIndex'");
+        waitForIndexQueryable();
+
```

Comment:
I have done some refactoring that does it.

Reply:
Hmm, I did not see it. Did you miss to push something, maybe? Just checking :-)

Reply:
It's in `AnalyzerEqOperatorSupportTest#assertIndexQueries`, [here](https://github.com/datastax/cassandra/blob/45eb18972d2bc0b9597d14bb04877ce1fbd57558/test/unit/org/apache/cassandra/index/sai/analyzer/AnalyzerEqOperatorSupportTest.java#L344-L346).

 I'm doing it manually instead of using `beforeAndAfterFlush` because for a single-line test call I prefer that if it fails the trace will point straight to the line before or after the flush. With `beforeAndAfterFlush` it's a bit harder to know if the failure happened before or after the flush.